evalml/automl/automl_search.py

```diff
@@ -100,6 +110,22 @@ def search(X_train=None, y_train=None, problem_type=None, objective="auto", **kw
     X_train = infer_feature_types(X_train)
     y_train = infer_feature_types(y_train)
     problem_type = handle_problem_types(problem_type)
+
+    datetime_column = None
```

Comment:
Depending on the outcome of our time series discussions, we can change this if we want to consider the index as the default placement of datetime data for time series. Until then, we'd need the user to pass in a minimum of `date_index` for time series default objectives.